### PR TITLE
proto: add health proto

### PR DESCRIPTION
**Pull request by BusyJay**
This pr adds health protos for the need to build health service.

Also protoc dependency is dropped in this pr, so when including grpcio-proto as dependency, user doesn't have to install protoc to make it work.

/cc pingcap/tikv#2290

### CODE REVIEW

**Comment by siddontang** [comment]
Do we need `proto/example.desc` and `proto/health.desc` ?

**Comment by BusyJay** [comment]
Yes. They are needed to generate source at compile time.

**Comment by siddontang** [comment]
The binary can be used in Windows or MacOSX?

**Comment by BusyJay** [comment]
Yes. Actually it's a proto message.

File: proto/proto/grpc/health/v1/health.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package grpc.health.v1;
+option csharp_namespace = "Grpc.Health.V1";
```
Comment (siddontang) [inline proto/proto/grpc/health/v1/health.proto]: do we need this?

**Comment by siddontang** [comment]
seem we should update the version too.

File: proto/proto/grpc/health/v1/health.proto (hunk 1 above)
Comment (BusyJay) [reply proto/proto/grpc/health/v1/health.proto]: The file is just copied from grpc repository, I prefer to leave it as it is.

**Comment by BusyJay** [comment]
Yes, it will be updated in a separate commit.

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
ping

**Pull request merged**